README.md
# React TVMaze

## Deployed App
https://git.generalassemb.ly/pages/ga-wdi-exercises/react-tvmaze/

## Starter Code:
react-express-webpack starter app: [https://github.com/wdi-sg/react-express-webpack/tree/tvmaze-react](https://github.com/wdi-sg/react-express-webpack/tree/tvmaze-react)

## [Start with a Mock](https://facebook.github.io/react/docs/thinking-in-react.html#start-with-a-mock)
First step in creating a React app is to start with a mock and some sample data.
These are the two views for our app:

### The Search Page
![search page](./images/search.png)

I've identified two components on the search page.
1. The top level component, which we'll call `Home`, is boxed in magenta.
2. The search input, a sub-component of `Home`, in yellow we'll call `Search`.

### The Results Page
![results page](./images/results.png)

I've identified three components on this page.
1. The same `Home` top level component.
2. A `results` components which contains results and an option to search again.
3. The individual results

### Component Hierarchy

Given these breakdowns we have a component hierarchy that looks like:
- `Home`
  - `Search`
  - `Results`
    - `Result`

#### Sample Data
Use `results.json` as the result you would get back from any search.

## [Build a Static Version of the App](https://facebook.github.io/react/docs/thinking-in-react.html#step-2-build-a-static-version-in-react)

First we will build a static version of the app passing all of our data by `props`.
This makes it much easier to avoid getting bogged down in tricky details of functionality while implementing the visual appearance of the UI.

## [Identify the Minimal Representation of UI State](https://facebook.github.io/react/docs/thinking-in-react.html#step-3-identify-the-minimal-but-complete-representation-of-ui-state)

For our app to work we need:
- `movies` (movies to show)
- `query` (title being searched)
- `hasSearched` (boolean determining wether to show the search input or the results)

All of these are subject to change over time and so each must be kept in state.

## [Identify Where Your State Should Live](https://facebook.github.io/react/docs/thinking-in-react.html#step-4-identify-where-your-state-should-live)

Central to considering where state lives is the idea of **one way data flow**.
The react documentation describes this step as "often the most challenging part for newcomers to understand".

Our task here is to look for the component for each aspect of state that could be the one place where that state is managed.

In our app, `query` is needed to keep track of what is going on in the search box, as well as to make the actual query.
This request will return the movies to the same component which managed the query so `movies` should be managed by the same component.
Finally, we have our `hasSearched` flag which we need to set when we make the request so these should all live in the same place.

Currently, the parent to the `Results` and `Search` components is `Home`.
We don't want to clutter our top level component as our app grows so this segues nicely into the idea of Container and Presentational Components.

### [Container & Presentational Components](https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0)
The above workflow has led to the currently very popular component architecture of distinguishing container and presentational components.

Presentational components are components that render themselves based solely on the information that they receive from props. At this point, all of our components are presentational.

Container components are components whose job it is to exclusively manage state and as props any data needed by its presentational components.

This leads to a very nice division where state management and presentation are cleanly separated.

We are going to create a `SearchContainer` to manage `query`, `shows`, and `hasSearched`.

## [Add Inverse Data Flow](https://facebook.github.io/react/docs/thinking-in-react.html#step-5-add-inverse-data-flow)
The last step is passing callbacks through props to presentational components to provide behavior.
We will need three functions defined on the `SearchContainer` component to provide necessary behavior to `Search` and `Results`:

1. `handleSearchInput` for managing changes to the input field
2. `onSubmitQuery` for kicking of the ajax request
3. `onSearchAgain` to set state back to render the search bar

> What is the point of all of these `.bind(this)` statements?

### Further
Create the ability to sort on different fields.

### Further
Link the show result to another part of the TVmaze api. By hand, download and include the results of that api search to use in your react app.


CHANGELOG.md
# Change Log

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

<a name="2.2.0"></a>
# [2.2.0](https://github.com/antonfisher/react-express-webpack/compare/v2.1.0...v2.2.0) (2018-03-12)


### Features

* hide package-lock.json from git diff output ([6b63ff2](https://github.com/antonfisher/react-express-webpack/commit/6b63ff2))
* upgrade to Webpack 4 ([f6224bc](https://github.com/antonfisher/react-express-webpack/commit/f6224bc))
* use compact build output ([5d5bb55](https://github.com/antonfisher/react-express-webpack/commit/5d5bb55))
* use last version of uglifyjs ([cd71f35](https://github.com/antonfisher/react-express-webpack/commit/cd71f35))
* use react-hot-loader v4 ([121bda7](https://github.com/antonfisher/react-express-webpack/commit/121bda7))
* use webpack merge for merging configs ([d765ec6](https://github.com/antonfisher/react-express-webpack/commit/d765ec6))



<a name="2.1.0"></a>
# [2.1.0](https://github.com/antonfisher/react-express-webpack/compare/v2.0.0...v2.1.0) (2018-02-01)


### Bug Fixes

* always include Intl to main bundle ([7b9a36f](https://github.com/antonfisher/react-express-webpack/commit/7b9a36f))


### Features

* update packadge.json for windows support (closes [#3](https://github.com/antonfisher/react-express-webpack/issues/3)) ([2b8f7ac](https://github.com/antonfisher/react-express-webpack/commit/2b8f7ac))



<a name="2.0.0"></a>
# [2.0.0](https://github.com/antonfisher/react-express-webpack/compare/v1.2.0...v2.0.0) (2018-01-27)


### Bug Fixes

* move html-webpack-harddisk-plugin dependency to dev ([9fe7747](https://github.com/antonfisher/react-express-webpack/commit/9fe7747))
* remove unused dependency 'history' ([75bc02f](https://github.com/antonfisher/react-express-webpack/commit/75bc02f))
* remove unused gif file-loader rule ([f121d0b](https://github.com/antonfisher/react-express-webpack/commit/f121d0b))
* use application logger in webpack-dev-middleware ([badfbc0](https://github.com/antonfisher/react-express-webpack/commit/badfbc0))


### Features

* remove react-tap-event-plugin dependency ([a6cd1e9](https://github.com/antonfisher/react-express-webpack/commit/a6cd1e9))
* support for realy old browsers, add intl polyfill (closes [#2](https://github.com/antonfisher/react-express-webpack/issues/2)) ([94f4e95](https://github.com/antonfisher/react-express-webpack/commit/94f4e95))
* update React to 16.2.0 ([4f9bd1b](https://github.com/antonfisher/react-express-webpack/commit/4f9bd1b))
* upgrade to react 16 ([1f73233](https://github.com/antonfisher/react-express-webpack/commit/1f73233))
* use flat componets/containers structure ([e44986b](https://github.com/antonfisher/react-express-webpack/commit/e44986b))
* use prettier code formatter ([894f014](https://github.com/antonfisher/react-express-webpack/commit/894f014))



<a name="1.2.0"></a>
# [1.2.0](https://github.com/antonfisher/react-express-webpack/compare/v1.1.0...v1.2.0) (2017-09-22)


### Bug Fixes

* don't sync router with redux store with old way ([70b0d1d](https://github.com/antonfisher/react-express-webpack/commit/70b0d1d))


### Features

* use '/' as root page url ([30b75b6](https://github.com/antonfisher/react-express-webpack/commit/30b75b6))



<a name="1.1.0"></a>
# [1.1.0](https://github.com/antonfisher/react-express-webpack/compare/v1.0.3...v1.1.0) (2017-09-22)


### Features

* add ip to logger output, don't log query starting ([88c589f](https://github.com/antonfisher/react-express-webpack/commit/88c589f))
* add unhandled errors handling, rename APP_PORT to HTTP_PORT, log app env ([8584748](https://github.com/antonfisher/react-express-webpack/commit/8584748))
* remove separated app config file ([ce0d66f](https://github.com/antonfisher/react-express-webpack/commit/ce0d66f))



<a name="1.0.2"></a>
## [1.0.2](https://github.com/antonfisher/react-express-webpack/compare/v1.0.0...v1.0.2) (2017-04-28)


### Bug Fixes

* always show response status code on error ([fcaf20b](https://github.com/antonfisher/react-express-webpack/commit/fcaf20b))

results.js
/*eslint-disable*/

module.exports = {
    results
};

var results = [
    {
        "score": 16.974672,
        "show": {
            "id": 139,
            "url": "http://www.tvmaze.com/shows/139/girls",
            "name": "Girls",
            "type": "Scripted",
            "language": "English",
            "genres": [
                "Drama",
                "Romance"
            ],
            "status": "Ended",
            "runtime": 30,
            "premiered": "2012-04-15",
            "officialSite": "http://www.hbo.com/girls",
            "schedule": {
                "time": "22:00",
                "days": [
                    "Sunday"
                ]
            },
            "rating": {
                "average": 6.7
            },
            "weight": 90,
            "network": {
                "id": 8,
                "name": "HBO",
                "country": {
                    "name": "United States",
                    "code": "US",
                    "timezone": "America/New_York"
                }
            },
            "webChannel": null,
            "externals": {
                "tvrage": 30124,
                "thetvdb": 220411,
                "imdb": "tt1723816"
            },
            "image": {
                "medium": "http://static.tvmaze.com/uploads/images/medium_portrait/31/78286.jpg",
                "original": "http://static.tvmaze.com/uploads/images/original_untouched/31/78286.jpg"
            },
            "summary": "<p>This Emmy winning series is a comic look at the assorted humiliations and rare triumphs of a group of girls in their 20s.</p>",
            "updated": 1519352733,
            "_links": {
                "self": {
                    "href": "http://api.tvmaze.com/shows/139"
                },
                "previousepisode": {
                    "href": "http://api.tvmaze.com/episodes/1079686"
                }
            }
        }
    },
    {
        "score": 13.338135,
        "show": {
            "id": 23542,
            "url": "http://www.tvmaze.com/shows/23542/good-girls",
            "name": "Good Girls",
            "type": "Scripted",
            "language": "English",
            "genres": [
                "Drama",
                "Comedy",
                "Crime"
            ],
            "status": "Running",
            "runtime": 60,
            "premiered": "2018-02-26",
            "officialSite": "https://www.nbc.com/good-girls?nbc=1",
            "schedule": {
                "time": "22:00",
                "days": [
                    "Monday"
                ]
            },
            "rating": {
                "average": 6.6
            },
            "weight": 99,
            "network": {
                "id": 1,
                "name": "NBC",
                "country": {
                    "name": "United States",
                    "code": "US",
                    "timezone": "America/New_York"
                }
            },
            "webChannel": null,
            "externals": {
                "tvrage": null,
                "thetvdb": 328577,
                "imdb": "tt6474378"
            },
            "image": {
                "medium": "http://static.tvmaze.com/uploads/images/medium_portrait/141/354343.jpg",
                "original": "http://static.tvmaze.com/uploads/images/original_untouched/141/354343.jpg"
            },
            "summary": "<p><b>Good Girls</b> follows three \"good girl\" suburban wives and mothers who suddenly find themselves in desperate circumstances and decide to stop playing it safe, and risk everything to take their power back.</p>",
            "updated": 1525293634,
            "_links": {
                "self": {
                    "href": "http://api.tvmaze.com/shows/23542"
                },
                "previousepisode": {
                    "href": "http://api.tvmaze.com/episodes/1425058"
                }
            }
        }
    },
    {
        "score": 13.020464,
        "show": {
            "id": 6771,
            "url": "http://www.tvmaze.com/shows/6771/the-powerpuff-girls",
            "name": "The Powerpuff Girls",
            "type": "Animation",
            "language": "English",
            "genres": [
                "Comedy",
                "Action",
                "Science-Fiction"
            ],
            "status": "Running",
            "runtime": 15,
            "premiered": "2016-04-04",
            "officialSite": "https://www.cartoonnetwork.com/video/powerpuff-girls/index.html",
            "schedule": {
                "time": "17:30",
                "days": [
                    "Sunday"
                ]
            },
            "rating": {
                "average": null
            },
            "weight": 90,
            "network": {
                "id": 11,
                "name": "Cartoon Network",
                "country": {
                    "name": "United States",
                    "code": "US",
                    "timezone": "America/New_York"
                }
            },
            "webChannel": null,
            "externals": {
                "tvrage": null,
                "thetvdb": 307473,
                "imdb": "tt4718304"
            },
            "image": {
                "medium": "http://static.tvmaze.com/uploads/images/medium_portrait/60/151357.jpg",
                "original": "http://static.tvmaze.com/uploads/images/original_untouched/60/151357.jpg"
            },
            "summary": "<p>The city of Townsville may be a beautiful, bustling metropolis, but don't be fooled! There's evil afoot! And only three things can keep the bad guys at bay: Blossom, Bubbles and Buttercup, three super-powered little girls, known to their fans (and villains everywhere) as <b>The Powerpuff Girls</b>. Juggling school, bedtimes, and beating up giant monsters may be daunting, but together the Powerpuff Girls are up to the task. Battling a who's who of evil, they show what it really means to \"fight like a girl.\"</p>",
            "updated": 1525638930,
            "_links": {
                "self": {
                    "href": "http://api.tvmaze.com/shows/6771"
                },
                "previousepisode": {
                    "href": "http://api.tvmaze.com/episodes/1454433"
                },
                "nextepisode": {
                    "href": "http://api.tvmaze.com/episodes/1454434"
                }
            }
        }
    },
    {
        "score": 12.998356,
        "show": {
            "id": 525,
            "url": "http://www.tvmaze.com/shows/525/gilmore-girls",
            "name": "Gilmore Girls",
            "type": "Scripted",
            "language": "English",
            "genres": [
                "Drama",
                "Comedy",
                "Romance"
            ],
            "status": "Ended",
            "runtime": 60,
            "premiered": "2000-10-05",
            "officialSite": null,
            "schedule": {
                "time": "21:00",
                "days": [
                    "Tuesday"
                ]
            },
            "rating": {
                "average": 8.8
            },
            "weight": 90,
            "network": {
                "id": 5,
                "name": "The CW",
                "country": {
                    "name": "United States",
                    "code": "US",
                    "timezone": "America/New_York"
                }
            },
            "webChannel": null,
            "externals": {
                "tvrage": 3683,
                "thetvdb": 76568,
                "imdb": "tt0238784"
            },
            "image": {
                "medium": "http://static.tvmaze.com/uploads/images/medium_portrait/4/11308.jpg",
                "original": "http://static.tvmaze.com/uploads/images/original_untouched/4/11308.jpg"
            },
            "summary": "<p><b>Gilmore Girls</b> is a drama centering around the relationship between a thirtysomething single mother and her teen daughter living in Stars Hollow, Connecticut.</p>",
            "updated": 1522264991,
            "_links": {
                "self": {
                    "href": "http://api.tvmaze.com/shows/525"
                },
                "previousepisode": {
                    "href": "http://api.tvmaze.com/episodes/47639"
                }
            }
        }
    },
    {
        "score": 12.963986,
        "show": {
            "id": 32087,
            "url": "http://www.tvmaze.com/shows/32087/chicken-girls",
            "name": "Chicken Girls",
            "type": "Scripted",
            "language": "English",
            "genres": [

            ],
            "status": "Running",
            "runtime": 16,
            "premiered": "2017-09-05",
            "officialSite": "https://www.youtube.com/playlist?list=PLVewHiZp3_LPhqzbcZFmS3iuDm9HymTsy",
            "schedule": {
                "time": "",
                "days": [
                    "Tuesday"
                ]
            },
            "rating": {
                "average": null
            },
            "weight": 85,
            "network": null,
            "webChannel": {
                "id": 21,
                "name": "YouTube",
                "country": null
            },
            "externals": {
                "tvrage": null,
                "thetvdb": null,
                "imdb": null
            },
            "image": {
                "medium": "http://static.tvmaze.com/uploads/images/medium_portrait/137/344365.jpg",
                "original": "http://static.tvmaze.com/uploads/images/original_untouched/137/344365.jpg"
            },
            "summary": "<p>Rhyme and her friends — known by their 'ship name, \"The Chicken Girls\" — have been dancing together forever. But this year, everything's changing...</p>",
            "updated": 1525391254,
            "_links": {
                "self": {
                    "href": "http://api.tvmaze.com/shows/32087"
                },
                "previousepisode": {
                    "href": "http://api.tvmaze.com/episodes/1439040"
                },
                "nextepisode": {
                    "href": "http://api.tvmaze.com/episodes/1439041"
                }
            }
        }
    },
    {
        "score": 12.147399,
        "show": {
            "id": 3418,
            "url": "http://www.tvmaze.com/shows/3418/anzac-girls",
            "name": "ANZAC Girls",
            "type": "Scripted",
            "language": "English",
            "genres": [
                "Drama",
                "War",
                "History"
            ],
            "status": "Ended",
            "runtime": 60,
            "premiered": "2014-08-10",
            "officialSite": "http://www.abc.net.au/tv/programs/anzac-girls/",
            "schedule": {
                "time": "20:30",
                "days": [
                    "Sunday"
                ]
            },
            "rating": {
                "average": 8.5
            },
            "weight": 0,
            "network": {
                "id": 114,
                "name": "ABC",
                "country": {
                    "name": "Australia",
                    "code": "AU",
                    "timezone": "Australia/Sydney"
                }
            },
            "webChannel": null,
            "externals": {
                "tvrage": 43688,
                "thetvdb": 284182,
                "imdb": "tt3078602"
            },
            "image": {
                "medium": "http://static.tvmaze.com/uploads/images/medium_portrait/17/44896.jpg",
                "original": "http://static.tvmaze.com/uploads/images/original_untouched/17/44896.jpg"
            },
            "summary": "<p>Honouring the Centenary of the commencement of WW1, <b>ANZAC Girls</b> is a moving new six-part series based on the unique, and rarely told true stories of Australian and New Zealand nurses serving at Gallipoli and the Western Front.</p>",
            "updated": 1477274719,
            "_links": {
                "self": {
                    "href": "http://api.tvmaze.com/shows/3418"
                },
                "previousepisode": {
                    "href": "http://api.tvmaze.com/episodes/226941"
                }
            }
        }
    },
    {
        "score": 12.13577,
        "show": {
            "id": 33320,
            "url": "http://www.tvmaze.com/shows/33320/derry-girls",
            "name": "Derry Girls",
            "type": "Scripted",
            "language": "English",
            "genres": [
                "Comedy"
            ],
            "status": "Running",
            "runtime": 30,
            "premiered": "2018-01-04",
            "officialSite": "http://www.channel4.com/programmes/derry-girls",
            "schedule": {
                "time": "22:00",
                "days": [
                    "Thursday"
                ]
            },
            "rating": {
                "average": 6.8
            },
            "weight": 90,
            "network": {
                "id": 45,
                "name": "Channel 4",
                "country": {
                    "name": "United Kingdom",
                    "code": "GB",
                    "timezone": "Europe/London"
                }
            },
            "webChannel": null,
            "externals": {
                "tvrage": null,
                "thetvdb": 338903,
                "imdb": "tt7120662"
            },
            "image": {
                "medium": "http://static.tvmaze.com/uploads/images/medium_portrait/141/353703.jpg",
                "original": "http://static.tvmaze.com/uploads/images/original_untouched/141/353703.jpg"
            },
            "summary": "<p>16-year-old Erin Quinn lives with her uncompromising mother, her long-suffering father and the fearsome ‘Granda Joe', a man whose love for his daughters and granddaughters is surpassed only by his contempt for his son-in-law.</p><p>It's the early 90s, and Erin is used to seeing her country on the nightly news and speaking in acronyms (The IRA, The UDA, The RUC). This is a time of armed police in armoured Land Rovers and British Army check points. But it's also the time of Murder She Wrote, The Cranberries, MJ and Lisa Marie, Doc Martens, bomber jackets, The X Files, Nirvana and Wayne's World. And while The Troubles may hang over her home town, Erin has troubles of her own</p>",
            "updated": 1518470029,
            "_links": {
                "self": {
                    "href": "http://api.tvmaze.com/shows/33320"
                },
                "previousepisode": {
                    "href": "http://api.tvmaze.com/episodes/1370222"
                }
            }
        }
    },
    {
        "score": 11.98947,
        "show": {
            "id": 1308,
            "url": "http://www.tvmaze.com/shows/1308/bad-girls",
            "name": "Bad Girls",
            "type": "Scripted",
            "language": "English",
            "genres": [
                "Drama",
                "Crime",
                "Romance"
            ],
            "status": "Ended",
            "runtime": 60,
            "premiered": "1999-06-01",
            "officialSite": null,
            "schedule": {
                "time": "21:00",
                "days": [
                    "Tuesday"
                ]
            },
            "rating": {
                "average": 9
            },
            "weight": 5,
            "network": {
                "id": 35,
                "name": "ITV",
                "country": {
                    "name": "United Kingdom",
                    "code": "GB",
                    "timezone": "Europe/London"
                }
            },
            "webChannel": null,
            "externals": {
                "tvrage": 2701,
                "thetvdb": 75328,
                "imdb": "tt0203248"
            },
            "image": {
                "medium": "http://static.tvmaze.com/uploads/images/medium_portrait/7/18492.jpg",
                "original": "http://static.tvmaze.com/uploads/images/original_untouched/7/18492.jpg"
            },
            "summary": "<p>Step inside HMP Larkhall and meet the women of G-Wing, who've committed various offences that range from shoplifting to murder. <b>Bad Girls</b> shows what life is like for these women in the 'inside world' and also shows the real-life problems (and maybe behaviour!) of the officers who watch them!</p>",
            "updated": 1483430334,
            "_links": {
                "self": {
                    "href": "http://api.tvmaze.com/shows/1308"
                },
                "previousepisode": {
                    "href": "http://api.tvmaze.com/episodes/111008"
                }
            }
        }
    },
    {
        "score": 11.941787,
        "show": {
            "id": 9136,
            "url": "http://www.tvmaze.com/shows/9136/funny-girls",
            "name": "Funny Girls",
            "type": "Scripted",
            "language": "English",
            "genres": [
                "Comedy"
            ],
            "status": "Running",
            "runtime": 30,
            "premiered": "2015-10-23",
            "officialSite": null,
            "schedule": {
                "time": "22:00",
                "days": [
                    "Friday"
                ]
            },
            "rating": {
                "average": null
            },
            "weight": 0,
            "network": {
                "id": 292,
                "name": "Three",
                "country": {
                    "name": "New Zealand",
                    "code": "NZ",
                    "timezone": "Pacific/Auckland"
                }
            },
            "webChannel": null,
            "externals": {
                "tvrage": null,
                "thetvdb": 302258,
                "imdb": null
            },
            "image": {
                "medium": "http://static.tvmaze.com/uploads/images/medium_portrait/32/80632.jpg",
                "original": "http://static.tvmaze.com/uploads/images/original_untouched/32/80632.jpg"
            },
            "summary": "<p><b>Funny Girls</b> is a shiny new sketch comedy show that is funny…and has girls! Featuring fast paced sketches that cover issues such as work, life, relationships and what to do when you're being haunted by a photobombing ghost. Set on a fictional television show where a large group of men produce an all-female sketch show. Rose Matafeo fronts the show alongside Laura Daniel and their friends including Jackie van Beek, who plays the show's producer, Pauline - who definitely has no experience in television.</p>",
            "updated": 1525556449,
            "_links": {
                "self": {
                    "href": "http://api.tvmaze.com/shows/9136"
                },
                "previousepisode": {
                    "href": "http://api.tvmaze.com/episodes/981043"
                },
                "nextepisode": {
                    "href": "http://api.tvmaze.com/episodes/1463593"
                }
            }
        }
    },
    {
        "score": 11.90232,
        "show": {
            "id": 7827,
            "url": "http://www.tvmaze.com/shows/7827/land-girls",
            "name": "Land Girls",
            "type": "Scripted",
            "language": "English",
            "genres": [
                "Drama",
                "Romance",
                "War"
            ],
            "status": "Ended",
            "runtime": 60,
            "premiered": "2009-09-07",
            "officialSite": "http://www.bbc.co.uk/programmes/b00xxnhv",
            "schedule": {
                "time": "17:15",
                "days": [
                    "Monday",
                    "Tuesday",
                    "Wednesday",
                    "Thursday",
                    "Friday"
                ]
            },
            "rating": {
                "average": null
            },
            "weight": 47,
            "network": {
                "id": 12,
                "name": "BBC One",
                "country": {
                    "name": "United Kingdom",
                    "code": "GB",
                    "timezone": "Europe/London"
                }
            },
            "webChannel": null,
            "externals": {
                "tvrage": null,
                "thetvdb": 112871,
                "imdb": "tt1449940"
            },
            "image": {
                "medium": "http://static.tvmaze.com/uploads/images/medium_portrait/28/71862.jpg",
                "original": "http://static.tvmaze.com/uploads/images/original_untouched/28/71862.jpg"
            },
            "summary": "<p>Follow the lives, loves and highs and lows of four members of the Women's Land Army who are working at the Hoxley Estate during World War II.</p><p><b>Land Girls</b> is a British television period drama series, first broadcast on BBC One on 7 September 2009. <i>Land Girls</i> was created by Roland Moore and commissioned by the BBC to commemorate the 70th anniversary of the outbreak of World War II. The programme was BBC Daytime's first commission of a period drama. <i>Land Girls</i> was filmed in and around the city of Birmingham. The first series features Summer Strallen, Christine Bottomley, Jo Woodcock and Becci Gemmell as four different girls doing their bit for Britain in theWomen's Land Army during the War.</p><p><i>Land Girls</i> won the \"Best Daytime Programme\" at the 2010 Broadcast Awards and in that same year the BBC announced that it had commissioned a second series, comprising five episodes. Woodcock and Gemmell reprised their roles as Bea and Joyce and Seline Hizlimade her debut as new girl, Connie Carter. The second series began airing from 17 January 2011 and two months later BBC Daytime Controller, Liam Keelan, renewed <i>Land Girls</i> for a third series. It began airing from 7 November 2011.</p>",
            "updated": 1524912549,
            "_links": {
                "self": {
                    "href": "http://api.tvmaze.com/shows/7827"
                },
                "previousepisode": {
                    "href": "http://api.tvmaze.com/episodes/437708"
                }
            }
        }
    }
]